client/src/main/java/io/split/client/SplitClientImpl.java

```diff
@@ -341,6 +371,86 @@ private Map<String, SplitResult> getTreatmentsWithConfigInternal(String matching
         }
     }
 
+    private Map<String, SplitResult> getTreatmentsWithSetsAndWithConfigInternal(String matchingKey, String bucketingKey, List<String> sets,
+                                                                     Map<String, Object> attributes, MethodEnum methodEnum) {
+        long initTime = System.currentTimeMillis();
+        if (sets == null || sets.isEmpty()) {
+            _log.warn(String.format("%s: sets must be a non-empty array", methodEnum.getMethod()));
+            return new HashMap<>();
+        }
+        FlagSetsValidResult flagSetsValidResult = FlagSetsValidator.areValid(sets);
+        try {
+            if (!flagSetsValidResult.getValid()) {
+                _log.warn("The sets are invalid");
+                return new HashMap<>();
+            }
+            if (filterSetsAreInConfig(flagSetsValidResult.getFlagSets()).isEmpty()) {
+                _log.warn("The sets are not");
+                return new HashMap<>();
+            }
+            checkSDKReady(methodEnum);
+            if (_container.isDestroyed()) {
+                _log.error("Client has already been destroyed - no calls possible");
+                return createMapControl(getAllFlags(flagSetsValidResult.getFlagSets()));
+            }
```

Comment:
I think the matchingKeyValidation is missing
```suggestion
            }
            
            if (!KeyValidator.isValid(matchingKey, "matchingKey", _config.maxStringLength(), methodEnum.getMethod())) {
            return createMapControl(getAllFlags(flagSetsValidResult.getFlagSets()));
        }
```

Reply:
Updated.